### PR TITLE
Remove energy efficient agent and related tests, documentation

### LINKED ISSUE #801: Missing description of region hints for energy efficiency agent in Man page
Refer: https://geopm.github.io/man/geopm_agent_energy_efficient.7.html

### LINKED ISSUE #2310: Remove the energy efficient agent
> estimate 1

We will bring back an EE agent in the future, but the current implementation doesn't do what it says on the tin, and the future agent may have a different policy interface.  Additionally, the agent is simply broken at runtime (see #2301).

Done criteria:
- [ ] Remove all documentation, tests and implementation for the energy efficient agent

### PR DESCRIPTION
**Pull request by avilcheslopez**
Signed-off-by: Alejandro Vilches <alejandro.vilches@intel.com>

- Relates to #2301
- Fixes #2310
- Fixes #801

### CODE REVIEW

**Comment by avilcheslopez** [comment]
NOTE: created as draft for now because I'm still looking into the appropriate changes for some the integration tests and tutorials.

File: integration/test/test_plugin_static_policy.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def skip_unless_energy_agent():
         agent = getSystemConfigAgent()
     except BaseException as ex:
         return unittest.skip('geopmadmin check failed: {}'.format(ex))
-    if agent not in ['frequency_map', 'energy_efficient']:
```
Comment (cmcantalupo) [inline integration/test/test_plugin_static_policy.py:73]: Lost the "not" qualifier, probably be better if this were a list of the two agents that require RAPL: "power_balancer" and "power_governor"

Comment (avilcheslopez) [reply integration/test/test_plugin_static_policy.py:73]: :o! Thanks for catching that! Will fix!

File: tutorial/admin/04_setup_default_frequency_map.sh [hunk 2]
```diff
@@ -49,6 +49,6 @@ echo "{\"GEOPM_AGENT\": \"energy_efficient\", \"GEOPM_POLICY\": \"$POLICY_FILE_P
 #   > srun geopmread MSR::PERF_CTL:FREQ board 0
 #   3700000000
 #
-# GEOPM jobs use EE agent with above policy; check report
+# GEOPM jobs use frequency map agent with above policy; check report
 #   > geopmlaunch srun -N1 -n1 --geopm-report=plugin_test.report -- geopmbench ~/short.conf > geopm_stdout 2>&1 && grep Policy plugin_test.report
-#   Policy: {"FREQ_MIN": 1200000000, "FREQ_MAX": 1700000000}
```
Comment (cmcantalupo) [inline tutorial/admin/04_setup_default_frequency_map.sh:54]: why did 1.7 GHz -> 1.8 GHz?

**Review by cmcantalupo — COMMENTED** [review]
Just a couple of small issues.  Thanks

File: tutorial/admin/04_setup_default_frequency_map.sh (hunk 2 above)
Comment (avilcheslopez) [reply tutorial/admin/04_setup_default_frequency_map.sh:54]: No special reason. This example was inspired by example 03. It obtains the CPU frequency sticker and subtracts 300MHz from it. It shows 1.8GHz as a hypothetical result. To make the examples different enough, I'll make this one simply set a fixed frequency of 1.7GHz (as the max that was being used before).

File: integration/test/test_plugin_static_policy.py (hunk 1 above)
Comment (avilcheslopez) [reply integration/test/test_plugin_static_policy.py:73]: Just to clarify, do you mean:
- "skip unless rapl agent" or...
- "skip if rapl agent" ?

**Review by cmcantalupo — APPROVED** [review]

**Pull request merged**